src/DefinitionResolver.php

```diff
@@ -667,13 +666,27 @@ public function resolveExpressionNodeToType($expr)
                 } else {
                     $classFqn = substr((string)$t->getFqsen(), 1);
                 }
-                $fqn = $classFqn . '->' . $expr->memberName->getText($expr->getFileContents());
+                $add = '->' . $expr->memberName->getText($expr->getFileContents());
                 if ($expr->parent instanceof Node\Expression\CallExpression) {
-                    $fqn .= '()';
+                    $add .= '()';
                 }
+                $fqn = $classFqn . $add;
                 $def = $this->index->getDefinition($fqn);
                 if ($def !== null) {
                     return $def->type;
+                } else {
+                    $classDef = $this->index->getDefinition($classFqn);
```

Comment:
yes, and it iterates the direct parents of `$classDef`. But where does it look at the grandparents?